File: AddAPearl.Domain/SubItem.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace AddAPearl.Domain
{
    public class SubItem
    {
        public int SubItemId { get; set; }

        public Item Item { get; set; }

        public int ItemId { get; set; }

        public Product Product { get; set; }

        public int ProductId { get; set; }

        public string SubItemName { get; set; }

        public decimal? Price { get; set; }

        public decimal PurchasePrice { get; set; }

        public DateTime? PurchaseDate { get; set; }

        public string Description { get; set; }

        public Customer Owner { get; set; }

        public int OwnerId { get; set; }

        public Customer Customer { get; set; }

        public int CustomerId { get; set; }

        public decimal? Rating { get; set; }
    }
}

File: AddAPearl.Web/src/app/shared/models/subitem.ts
import {
  ICustomer,
  IItem,
  IProduct,
} from './index';

/* Defines the Item entity */
export interface ISubItem {
  subItemId: number;
  itemId: number;
  item: IItem;
  productId: number;
  product: IProduct;
  subItemName: string;
  price: number;
  purchasPrice: number;
  purchasDate: string;
  description: string;
  ownerId: number;
  owner: ICustomer;
  customerId: number;
  customer: ICustomer;
  rating: number;
}

export class SubItem implements ISubItem {
  constructor(
    public subItemId: number,
    public itemId: number,
    public item: IItem,
    public productId: number,
    public product: IProduct,
    public subItemName: string,
    public price: number,
    public purchasPrice: number,
    public purchasDate: string,
    public description: string,
    public ownerId: number,
    public owner: ICustomer,
    public customerId: number,
    public customer: ICustomer,
    public rating: number,
  ) {
    // Any other stuff to do during construction...
  }
}
